hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -609,16 +609,20 @@ public ListKeysResult listKeys(String volumeName, String bucketName,
       int maxKeys) throws IOException {
     Preconditions.checkNotNull(volumeName);
     Preconditions.checkNotNull(bucketName);
-
+    OmBucketInfo omBucketInfo = metadataManager.validateBucket(volumeName, bucketName);
+    BucketLayout bucketLayout = omBucketInfo.getBucketLayout();
     // We don't take a lock in this path, since we walk the
     // underlying table using an iterator. That automatically creates a
     // snapshot of the data, so we don't need these locks at a higher level
     // when we iterate.
-
-    if (enableFileSystemPaths) {
+    if (bucketLayout.shouldNormalizePaths(enableFileSystemPaths)) {
       startKey = OmUtils.normalizeKey(startKey, true);
       keyPrefix = OmUtils.normalizeKey(keyPrefix, true);
     }
+    /*startKey = OMClientRequest
```

Comment:
This line seems unintentional, could you please address this if not needed?

Reply:
Thanks @tanvipenumudy for review. I have removed the commented code.